lib/vagrant-service-manager/adb_executor.rb

```diff
@@ -0,0 +1,40 @@
+module VagrantPlugins
+  module ServiceManager
+    VERSION_CMD = {
+      docker: "docker version --format '{{.Server.Version}}'",
+      openshift: "oc version | grep 'oc' | grep -oE '[0-9.]+'"
+    }
+
+    class ADBExecutor
+      LABEL = 'servicemanager.commands.install_cli.message'
+
+      def initialize(binary_type, machine, ui, options)
+        @type = binary_type
+        @machine = machine
+        @ui = ui
+        @options = options
+      end
+
+      def execute
+        binary_exists = true
+        version = @options['--cli-version']
+        path = @options['--path']
+
+        version = PluginUtil.execute_once(@machine, @ui, VERSION_CMD[@type.to_sym]) if version.nil?
```

Comment:
Its a module so we need to invoke its method using module name. The way you describe could be achieved by converting into class and making other class subclass of it which I don't think good idea.
For example, think about scenarios where I have to use only one or two method of this plugin util class in `plugins/guests/cap` files. There it is not point to make it super class.

One thing we can do is we can filter few common methods which should be extracted into base class which can be subclass within the `lib/` folder appropriate classes.